### PR TITLE
[CDAP-2474] Change the ConnectionFactory JNDI name to be consistent with other properties in JMS source.

### PR DESCRIPTION
**Pull request by henry-cask**
Currently the name of the ConnectionFactory property for JMS source is called "ConnectionFactory".

Need to change it something to follow the rest of JMS source property names, for example: jms.destination.name

### CODE REVIEW

**Comment by henry-cask** [comment]
Build:
http://builds.cask.co/browse/CDAP-DUT1802-1

**Comment by awholegunch** [comment]
Well, where is JMS_CONNECTION_FACTORY_NAME now being used? & why didn't you add another string so as to not repeat yourself?

**Comment by hsaputra** [comment]
It is used in the JmsPluginConfig field name, not shown in the diff bc it was already there.
Adding another String constant is a good idea, will add it.

**Comment by henry-cask** [comment]
Build 2: http://builds.cask.co/browse/CDAP-DUT1802-2

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java [hunk 1]
```diff
@@ -64,7 +64,9 @@
   public static final String JMS_MESSAGES_TO_RECEIVE = "jms.messages.receive";
   public static final String JMS_NAMING_FACTORY_INITIAL = "jms.factory.initial";
   public static final String JMS_PROVIDER_URL = "jms.provider.url";
-  public static final String JMS_CONNECTION_FACTORY_NAME = "ConnectionFactory";
+  public static final String JMS_CONNECTION_FACTORY_NAME = "jms.jndi.connectionfactory.name";
+
+  public static final String CONNECTION_FACTORY = "ConnectionFactory";
```
Comment (bdmogal) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: Can we call this ``DEFAULT_CONNECTION_FACTORY_NAME``?

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java [hunk 2]
```diff
@@ -328,7 +330,7 @@ public JmsProvider getJmsProvider() {
 
     public JmsPluginConfig() {
       messagesToReceive = 50;
-      connectionFactoryName = "ConnectionFactory";
+      connectionFactoryName = CONNECTION_FACTORY;
     }
 
     public JmsPluginConfig(String destinationName, @Nullable Integer messagesToReceive, String initialContextFactory,
```
Comment (bdmogal) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: better to have ``@Nullable`` params at the end?

**Comment by bdmogal** [comment]
only a couple of minor comments

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java (hunk 1 above)
Comment (henry-cask) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: This is a string constant to just save some time to retype so usually the variable name is exactly the same as the String it represents

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java (hunk 2 above)
Comment (henry-cask) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: Will make the change

Comment (henry-cask) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: Actually this is already there and not part of the PR, I could send another one to change it.

Comment (bdmogal) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: ok.

**Comment by bdmogal** [comment]
Just the one comment to change the variable name to ``DEFAULT_CONNECTION_FACTORY``. LGTM otherwise

**Comment by henry-cask** [comment]
Thanks @bdmogal , I just updated the PR with your suggestion
